src/content/en/2020/javascript.md

```diff
@@ -533,27 +575,58 @@ Looking specifically at the frameworks in use, we see that the median JavaScript
+  sheets_gid="160832134",
+  sql_file="main_thread_time_frameworks.sql"
+) }}
 
 Tools like React, GSAP, and RequireJS tend to spend a lot of time on the main thread of the browser, regardless of whether it's a desktop or mobile page view. The same tools that tend to lead to less code overall—tools like Alpine and Svelte—also tend to lead to lower impact on the main thread.
 
 The gap between the experience a framework provides for desktop and mobile is also worth digging into. Mobile traffic is becoming increasingly dominant, and it's critical that our tools perform as well as possible for mobile pageviews. The bigger the gap we see between desktop and mobile performance for a framework, the bigger the red flag.
 
-{# Table? Gap in absolute and relative terms #}
+{{ figure_markup(
+  image="frameworks-main-thread-no-ember-diff.png",
+  caption="Difference between desktop and mobile median main thread time per page by JavaScript framework, excluding Ember.js.",
+  description="Bar chart showing the absolute and relative differences between desktop and mobile's median main thread time per page by JavaScript framework, excluding Ember.js. Polymer jumps out later in the popularity list as having a high difference: about 5 seconds and 250% slower median main thread time on mobile pages than desktop pages. Other frameworks that stand out are GSAP and RequireJS has having a 4 second or 150% difference. Frameworks with the lowest difference are Mustache and RxJS, which are only about 20-30% slower on mobile.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=1758266664&format=interactive",
+  sheets_gid="160832134",
+  sql_file="main_thread_time_frameworks.sql"
+) }}
+
+As you would expect, there's a gap for all tools in use due to the lower processing power of the [emulated Moto G4](methodology#webpagetest). Ember and Polymer seem to jump out as particularly egregious examples, while tools like RxJS and Mustache vary only minorly from desktop to mobile.
 
-As you would expect, there's a gap for all tools in use due to the lower processing power of the emulated Moto G4's. Ember and Polymer seem to jump out as particularly egregious examples, while tools like RxJS and Mustache vary only minorly from desktop to mobile.
+## What's the impact?
 
-## What's the Impact?
 We have a pretty good picture now of how much JavaScript we use, where it comes from and what we use it for. While that's interesting enough on its own, the real kicker is the "so what?" What impact does all this script actually have on the experience of our pages?
 
 The first thing we should consider is what happens with all that JavaScript once its been downloaded. Downloading is only the first part of the JavaScript journey. The browser still has to parse all that script, compile it, and eventually execute it. While browsers are increasingly finding ways to offload
```

Comment:
> While browsers are increasingly finding ways to offload

This sentence continues to be incomplete, I remember reporting it earlier.

Reply:
Yes, just got to that part of the edit and left a TODO

Reply:
Fixed! Sorry for missing that the first time around @ibnesayeed!